test/test_forms.jl

```diff
@@ -48,4 +51,19 @@ using Random
 
   @test underlying_graph(ket_network(qf)) == underlying_graph(ψket)
   @test underlying_graph(operator_network(qf)) == underlying_graph(A)
+
+  ∂qf_∂v = only(derivative_state(qf, [v]))
+  @test (∂qf_∂v) * (qf[ket_vertex_map(qf)(v)] * qf[bra_vertex_map(qf)(v)]) ≈
+    ITensors.contract(qf)
+
+  ∂qf_∂v_bp = derivative_state(qf, [v]; alg="bp", update_bp_cache=false)
+  ∂qf_∂v_bp = ITensors.contract(∂qf_∂v_bp)
+  ∂qf_∂v_bp = normalize!(∂qf_∂v_bp)
+  ∂qf_∂v = normalize!(∂qf_∂v)
```

Comment:
Let's use `normalize` instead of `normalize!`, same down below.

Reply:
I believe `normalize(T::ITensor)` is not defined and only the in-place version exists. I can do `∂qf_∂v /= norm(∂qf_∂v)` instead